### PR TITLE
Remember pageloader in ViewModel

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt [hunk 1]
```diff
@@ -108,11 +111,13 @@ private fun Background(
...
-            pageLoader.start()
-            onDispose {
-                pageLoader.stop()
+    Await(with(asyncInVM { preparePageLoader(args) }) { { await() } }) { loader ->
+        launchInVM {
+            loader.start()
+            try {
+                awaitCancellation()
+            } finally {
+                loader.stop()
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt:120]: If the use opens an external archive, the old archive in reader won't be replaced.

Comment (FooIbar) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt:120]: We need to cancel & relunch when `args` changes.

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt:121]: Done

**Review by FooIbar — APPROVED** [review]
Not gonna block this, but we should launch `setOrientation` etc in vm as well

**Pull request merged**